src/compiler/style/css-imports.ts

```diff
@@ -14,112 +27,104 @@ export const parseCssImports = async (
   resolvedFilePath: string,
   styleText: string,
   styleDocs?: d.StyleDoc[]
-) => {
+): Promise<ParseCSSReturn> => {
   const isCssEntry = resolvedFilePath.toLowerCase().endsWith('.css');
   const allCssImports: string[] = [];
-  const concatStyleText = await updateCssImports(
-    config,
-    compilerCtx,
-    buildCtx,
-    isCssEntry,
-    srcFilePath,
-    resolvedFilePath,
-    styleText,
-    allCssImports,
-    new Set(),
-    styleDocs
-  );
+
+  // a Set of reviously-resolved file paths that we add to as we traverse the
+  // import tree (to avoid a possible circular dependency and infinite loop)
+  const resolvedFilePaths = new Set();
+
+  const concatStyleText = await resolveAndFlattenImports(srcFilePath, resolvedFilePath, styleText);
+
   return {
     imports: allCssImports,
     styleText: concatStyleText,
   };
-};
-
-const updateCssImports = async (
-  config: d.Config,
-  compilerCtx: d.CompilerCtx,
-  buildCtx: d.BuildCtx,
-  isCssEntry: boolean,
-  srcFilePath: string,
-  resolvedFilePath: string,
-  styleText: string,
-  allCssImports: string[],
-  noLoop: Set<string>,
-  styleDocs?: d.StyleDoc[]
-) => {
-  if (noLoop.has(resolvedFilePath)) {
-    return styleText;
-  }
-  noLoop.add(resolvedFilePath);
 
-  if (styleDocs != null) {
-    parseStyleDocs(styleDocs, styleText);
-  }
+  /**
+   * Resolve and flatten all imports for a given CSS file, recursively crawling
+   * the tree of imports to resolve them all and produce a concatenated
+   * stylesheet. We declare this function here, within `parseCssImports`, in order
+   * to get access to `compilerCtx`, `buildCtx`, and more without having to pass
+   * a whole bunch of arguments.
+   *
+   * @param srcFilePath the source filepath
+   * @param resolvedFilePath the resolved filepath
+   * @param styleText style text we start with*
+   * @returns concatenated styles assembled from the various imported stylesheets
+   */
+  async function resolveAndFlattenImports(
```

Comment:
this function is a somewhat changed (and renamed) version of the `updateCssImports` function. I moved it inside of `parseCssImports` in order to get access to the arguments passed in to `parseCssImports` like `buildCtx` without having to pass them around as arguments (i.e. accessing them by just being in `parseCssImport`'s scope instead of passing arguments)

Reply:
I should note, I changed things around a bit but didn't mess with the core logic here